sharktank/sharktank/models/t5/t5.py

```diff
@@ -0,0 +1,1136 @@
+# Copyright 2018 Mesh TensorFlow authors, T5 Authors and HuggingFace Inc. team.
+# Copyright 2024 Advanced Micro Devices, Inc.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""T5 LLM adapted from transformers
+https://github.com/huggingface/transformers/blob/v4.40-release/src/transformers/models/t5/modeling_t5.py
+"""
+
+from typing import Any, Optional, Tuple
+from dataclasses import dataclass, field
+import math
+import torch
+from torch import nn
+import copy
+import logging
+import warnings
+from collections import OrderedDict
+
+from ...layers import (
+    BaseLayer,
+    RMSNormLayer,
+    TokenEmbeddingLayer,
+    LinearLayer,
+)
+from ... import ops
+from ...types.theta import Theta
+from ...layers import FFN
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class T5Config:
```

Comment:
This might need to move to `layers/configs/t5_configs.py`. Even better if moved to `llm_configs.py`

Reply:
I moved it there.